assets/js/tutorials/trade-in-the-dex.js

```diff
@@ -0,0 +1,278 @@
+
+function update_exchange_rate(event) {
+  const tpa = $("#taker-pays-amount-1").val()
+  const tga = $("#taker-gets-amount-1").val()
+  const exchange_rate = BigNumber(tga) / BigNumber(tpa)
+  $("#exchange-rate-1").val(`${exchange_rate}`)
+}
+$("#taker-pays-amount-1").change(update_exchange_rate)
+$("#taker-gets-amount-1").change(update_exchange_rate)
+
+$("#look-up-offers").click( async (event) => {
+  const block = $(event.target).closest(".interactive-block")
+  block.find(".output-area").html("")
+  const address = get_address(event)
+  if (!address) {return}
+
+  const we_want = {
+    "currency": $("#taker-pays-currency-1").val(),
+    "issuer": $("#taker-pays-issuer-1").val(),
+    "value": $("#taker-pays-amount-1").val()
+  }
+  const we_spend = {
+    "currency": "XRP",
+    "value": xrpl.xrpToDrops($("#taker-gets-amount-1").val())
+  }
+  const proposed_quality = BigNumber(we_spend.value) / BigNumber(we_want.value)
+  let orderbook_resp
+  try {
+    orderbook_resp = await api.request({
+      "command": "book_offers",
+      "taker": address,
+      "ledger_index": "current",
+      "taker_gets": we_want,
+      "taker_pays": we_spend
+    })
+    block.find(".output-area").append(
+      `<pre><code>${pretty_print(orderbook_resp.result)}`)
+  } catch(err) {
+    show_error(block, err)
+    block.find(".loader").hide()
+    return
+  }
+
+  // Estimate whether a proposed Offer would execute immediately, and...
+  // If so, how much of it? (Partial execution is possible)
+  // If not, how much liquidity is above it? (How deep in the order book would
+  //    other Offers have to go before ours would get taken?)
+  // Note: these estimates can be thrown off by rounding if the token issuer
+  // uses a TickSize setting other than the default (15). In that case, you
+  // can increase the TakerGets amount of your final offer just a little bit to
+  // compensate.
+
+  const offers = orderbook_resp.result.offers
+  const want_amt = BigNumber(we_want.value)
+  let running_total = BigNumber(0)
+  if (!offers) {
+    block.find(".output-area").append(`<p>No Offers in the matching book.
+                 Offer probably won't execute immediately.</p>`)
+  } else {
+    for (const o of offers) {
+      if (o.quality <= proposed_quality) {
+        block.find(".output-area").append(
+            `<p>Matching Offer found, funded with ${o.owner_funds}</p>`)
+        running_total = running_total.plus(BigNumber(o.owner_funds))
+        if (running_total >= want_amt) {
+          block.find(".output-area").append("<p>Full Offer will probably fill.</p>")
+          break
+        }
+      } else {
+        // Offers are in ascending quality order, so no others after this
+        // will match, either
+        block.find(".output-area").append(`<p>Remaining orders too expensive.</p>`)
+        break
+      }
+    }
+    block.find(".output-area").append(`<p>Total matched: ${Math.min(running_total, want_amt)}</p>`)
```

Comment:
Should this include the units? (Or be a percentage?) - Otherwise it might be confusing which currency is being talked about.